### ISSUE TITLE
MVM: Address sky_footprint.build error

### ISSUE THREAD

**Issue by stscijgbot-hstdp** [spacetelescope/drizzlepac#1083]
_Issue [HLA-603](https://jira.stsci.edu/browse/HLA-603) was created on JIRA by [Warren Hack](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hack):_

The following error occurred while processing some (not all) MVM sky cells:

> 2021204165435 INFO src=drizzlepac.hapmultisequencer- Preparing configuration parameter values for filter product hst_skyce
> ell-p0081x16y14_acs_wfc_f850lp_all_drc.fits
> ^G^G^G
> Traceback (most recent call last):
>   File "/home/mburger/anaconda3/envs/caldp/lib/python3.7/site-packages/drizzlepac/hapmultisequencer.py", line 236, in run_mvm_processing
>     _ = filter_item.generate_metawcs()
>   File "/home/mburger/anaconda3/envs/caldp/lib/python3.7/site-packages/drizzlepac/haputils/product.py", line 1010, in generate_metawcs
>     mvm_footprint.build(self.all_mvm_exposures)
>   File "/home/mburger/anaconda3/envs/caldp/lib/python3.7/site-packages/drizzlepac/haputils/cell_utils.py", line 347, in build
>     blank = np.array(img).astype(np.int16)
> TypeError: int() argument must be a string, a bytes-like object or a number, not 'Image'
> 2021204165524 INFO src=drizzlepac.hapmultisequencer- Processing completed at 2021-07-23 16:55:24.115741

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-603?focusedCommentId=541239#comment-541239):_

```
No. images   Dataset
  6 p1008x18y18/poller_p1008x18y18.out
  7 p1713x18y07/poller_p1713x18y07.out
  7 p1713x19y07/poller_p1713x19y07.out
  9 p1958x11y18/poller_p1958x11y18.out
 20 p0400x14y11/poller_p0400x14y11.out
 29 p0799x12y15/poller_p0799x12y15.out
 33 p0081x16y13/poller_p0081x16y13.out
 45 p0799x13y15/poller_p0799x13y15.out
 81 p1164x17y13/poller_p1164x17y13.out
 88 p1164x15y15/poller_p1164x15y15.out
 93 p0410x14y05/poller_p0410x14y05.out
101 p2506x05y06/poller_p2506x05y06.out
101 p2506x06y06/poller_p2506x06y06.out
111 p1974x09y01/poller_p1974x09y01.out
117 p1164x17y14/poller_p1164x17y14.out
133 p1164x16y15/poller_p1164x16y15.out
208 p1164x15y14/poller_p1164x15y14.out
396 p1164x15y13/poller_p1164x15y13.out
737 p0081x15y14/poller_p0081x15y14.out    X
778 p0081x16y14/poller_p0081x16y14.out    X
831 p1164x16y14/poller_p1164x16y14.out
942 p1164x16y13/poller_p1164x16y13.out    X```

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-603?focusedCommentId=541244#comment-541244):_

The table above lists the number of images in each MVM dataset and the dataset name.  The "X" symbol marks the three datasets which failed both the 23 July and 26 July test runs.  The previous two test runs (19 and 20 July) were comprised of different datasets with the largest having 305 images.  At first blush it looks like a resources issue (e.g., memory), particularly as the error message seems odd for the circumstances:

TypeError: int() argument must be a string, a bytes-like object or a number, not 'Image'

The error occurs at line 347 in cell_utils.py: blank = np.array(img).astype(np.int16)

The p1164x16y14 dataset does process correctly (double checked), and this goes against the sheer resources issue.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Warren Hack](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hack) on [JIRA](https://jira.stsci.edu/browse/HLA-603?focusedCommentId=541327#comment-541327):_

Looking at the processing log for 'p1164x16y14', it is clear that the processing stopped midway without error while in the process of running AstroDrizzle to create one of the desired products, the second one if I remember correctly.  This would support a resource issue being the reason for why not all the products were created for this SkyCell since only 1 MVM product ended up being drizzled.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-603?focusedCommentId=541703#comment-541703):_

The bug for p0081x16y14 was caused by an image in the list not really overlapping the skycell.  This fix protects against this instance and is out for review as Git PR # 1088.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-603?focusedCommentId=541807#comment-541807):_

GIt PR#&#x2060;1008 which addressed this problem has been approved and merged.

**Comment by stsci-hack** [comment]
Closed by #1008.
